app/src/main/java/org/listenbrainz/android/ui/screens/settings/SettingsActivity.kt

```diff
@@ -66,18 +67,28 @@ class SettingsActivity : AppCompatActivity() {
             
             // Explicit Ui Mode functionality.
             if (preference.key == PREFERENCE_SYSTEM_THEME){
+
+                 val intent = Intent(this@SettingsActivity, DashboardActivity::class.java)
+                 intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
                 when (newValue) {
                     "Dark" -> {
                         setDefaultNightMode(MODE_NIGHT_YES)
                         isUiModeIsDark.value = true
+                        startActivity(intent)
```

Comment:
Why don't we move everything below the when clause and above the return statement?

Reply:
I want to run this two lines only when user makes some changes,like when they create a newValue(whatever it may be). see, I don't do it on else part. 
For these reason
> Why don't we move everything below the when clause and above the return statement?